file_snapshot.go

```diff
@@ -384,6 +384,11 @@ func (s *FileSnapshotSink) Close() error {
 	// Close the open handles
 	if err := s.finalize(); err != nil {
 		s.logger.Printf("[ERR] snapshot: Failed to finalize snapshot: %v", err)
+		s.logger.Printf("[INFO] snapshot: Deleting temporary files...")
```

Comment:
I ended up moving it here because finalize() is where it tries to save files to the temporary directory. If that fails, we can clean up after ourselves by deleting the temp directory right there.